reference/snapshot.md

```diff
@@ -0,0 +1,279 @@
+EOF
+```
+
+{% endtab %}
+
+{% tab title="YAML" %}
+
+```text
+kind: VolumeSnapshotClass
+apiVersion: snapshot.storage.k8s.io/v1
+metadata:
+  name: csi-mayastor-snapshotclass
+  annotations:
+    snapshot.storage.kubernetes.io/is-default-class: "true"
+driver: io.openebs.csi-mayastor
+deletionPolicy: Delete
+
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+
+
+| Parameters | Type | Description|
+| -------- | ---- | -------- |
+| **Name**| String | Custom name of the snapshot class|
+| **Driver** | String | CSI provisioner of the storage provider being requested to create a snapshot (io.openebs.csi-mayastor)|
+
+**Apply VolumeSnapshotClass details**
+
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+kubectl apply -f class.yaml
+```
+{% endtab %}
+
+{% tab title="Example Output" %}
+```text
+volumesnapshotclass.snapshot.storage.k8s.io/csi-mayastor-snapshotclass created
+```
+{% endtab %}
+{% endtabs %}
+
+
+
+
+### Step 2: Create the snapshot
+
+**Create a file to define a snapshot**
+
+{% tabs %}
+
+
+{% tab title="Command" %}
+
+```text=
+cat <<EOF | kubectl create -f -
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: mayastor-pvc-snap-1
+spec:
+  volumeSnapshotClassName: csi-mayastor-snapshotclass
+  source:
+    persistentVolumeClaimName: ms-volume-claim   
+EOF
+```
+{% endtab %}
+
+{% tab title="YAML" %}
+
+```text
+apiVersion: snapshot.storage.k8s.io/v1
+kind: VolumeSnapshot
+metadata:
+  name: mayastor-pvc-snap-1
+spec:
+  volumeSnapshotClassName: csi-mayastor-snapshotclass
+  source:
+    persistentVolumeClaimName: ms-volume-claim   
+
+```
+
+{% endtab %}
+
+{% endtabs %}
+
+
+| Parameters | Type | Description|
+| -------- | ---- | -------- |
+| **Name** | String | Name of the snapshot | 
+| **VolumeSnapshotClassName** | String | Name of the created snapshot class | 
+| **PersistentVolumeClaimName** | String | Name of the PVC. Example- `ms-volume-claim` | 
+
+
+**Apply the snapshot**
+
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+kubectl apply -f snapshot.yaml
+```
+{% endtab %}
+
+{% tab title="Example Output" %}
+```text
+volumesnapshot.snapshot.storage.k8s.io/mayastor-pvc-snap-1 created
+```
+{% endtab %}
+{% endtabs %}
+
+---------
+
+## List Snapshots 
+
+To retrieve the details of the created snapshots, use the following command:
+
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+kubectl get volumesnapshot 
+```
+{% endtab %}
+
+{% tab title="Example Output" %}
+```text
+NAME                READYTOUSE   SOURCEPVC         SOURCESNAPSHOTCONTENT                RESTORESIZE                            SNAPSHOTCLASS    SNAPSHOTCONTENTCREATIONTIME    AGE
+mayastor-pvc-snap-1   false     ms-volume-claim     1Gi csi-mayastor-snapshotclass  snapcontent-88aa857e-3c07-4ad9-b4f8-2a9d    2f209a4d              4m5s                   4m5s
+```
+{% endtab %}
+{% endtabs %}
+
+    
+:::note
+In the upcoming version, READYTOUSE field will be set to **true** after adding the support for cloning.
```

Comment:
Possible rephrase
The READYTOUSE field will be set as **true** in the upcoming version which will have the support to create a **clone** from the snapshot.